pydantic/types.py

```diff
@@ -3007,3 +3008,31 @@ def __get_pydantic_core_schema__(cls, source_type: Any, handler: GetCoreSchemaHa
 That is, instead of a [`ValidationError`][pydantic_core.ValidationError] being propagated up and the entire iterable being discarded
 any invalid items are discarded and the valid ones are returned.
 """
+
+
+@_dataclasses.dataclass(**_internal_dataclass.slots_true, frozen=True)
+class FailFast:
+    """A `FailFast` can be used to specify that validation should stop at the first error.
+
+    This can be useful when you want to validate a large amount of data and you only need to know if it's valid or not.
```

Comment:
Let's add a note on performance here.

Given that this is a performance focused feature, let's also add a note on this to the `Performance` section of the conceptual docs.

Reply:
What exactly should I add about performance? Do you smth in mind?

Reply:
Just that you might want to enable this setting if you want to validate your data faster (basically, if you use this, validation will be more performant with the caveat that you get less information).

In a sense, I think we should use `performance` as a buzzword where applicable bc people generally really appreciate the callout.

Reply:
I'm also happy to add these performance docs later if you'd prefer :).